kratos/solving_strategies/schemes/residual_based_bossak_velocity_scheme.h

```diff
@@ -0,0 +1,793 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:   Jordi Cotela
+//                  Suneth Warnakulasuriya
+//
+
+#if !defined(KRATOS_RESIDUAL_BASED_BOSSAK_VELOCITY_SCHEME_H_INCLUDED)
+#define KRATOS_RESIDUAL_BASED_BOSSAK_VELOCITY_SCHEME_H_INCLUDED
+
+// System includes
+#include <vector>
+
+// External includes
+
+// Project includes
+#include "includes/checks.h"
+#include "includes/define.h"
+#include "includes/kratos_parameters.h"
+#include "solving_strategies/schemes/scheme.h"
+#include "utilities/derivatives_extension.h"
+#include "utilities/indirect_scalar.h"
+
+namespace Kratos
+{
+///@name Kratos Classes
+///@{
+
+/// A scheme for dynamic adjoint equations, using Bossak time integration.
+/**
+ * It can be used for either first- or second-order time derivatives. Elements
+ * and conditions must provide a specialization of AdjointExtensions via their
+ * data value container, which allows the scheme to operate independently of
+ * the variable arrangements in the element or condition.
+ */
+template <class TSparseSpace, class TDenseSpace>
+class ResidualBasedBossakVelocityScheme : public Scheme<TSparseSpace, TDenseSpace>
```

Comment:
some general comments abt this class:
- It seems to be very specific to fluid, which is not a problem, just sad we cannot reuse more code :)
- Please take a look at `utilities/time_discretization` and use the functionalities provided there to avoid copy-pasting of code
- Do you really need the neighbor-elements-stuff? I think this is a pain in MPI and hence difficult to port
- Have you thought how to use this in MPI?

Reply:
+1
> Please take a look at utilities/time_discretization and use the functionalities provided there to avoid copy-pasting of code

Reply:
why specific of the fluid? this is in the core...it should be usable for everryhing